src/coreclr/tools/Common/Compiler/InstructionSetSupport.cs

```diff
@@ -333,12 +334,43 @@ public bool ComputeInstructionSetFlags(int maxVectorTBitWidth,
             {
                 unsupportedInstructionSets.AddInstructionSet(instructionSetConversion[unsupported]);
             }
+
             unsupportedInstructionSets.ExpandInstructionSetByReverseImplication(_architecture);
             unsupportedInstructionSets.Set64BitInstructionSetVariants(_architecture);
 
             if ((_architecture == TargetArchitecture.X86) || (_architecture == TargetArchitecture.ARM))
                 unsupportedInstructionSets.Set64BitInstructionSetVariantsUnconditionally(_architecture);
 
+            // While it's possible to enable individual AVX-512 ISA's, it is not
+            // optimal to do so, since they aren't totally functional this way,
+            // plus it is extremely rare to encounter hardware that doesn't support
+            // all of them. So, here we ensure that we are enabling all the ISA's
+            // if one is specified in the Crossgen2 or ILC command-lines.
+            //
+            // For more information, check this Github comment:
+            // https://github.com/dotnet/runtime/issues/106450#issuecomment-2299504035
+
+            if (_supportedInstructionSets.Any(iSet => iSet.Contains("avx512")))
```

Comment:
If we've gotten here, anything in `iSet` is known valid ISAs, right?

That is, we won't get here for something like `avx512typo` or anything?

Reply:
Actually, this is fine. Prior to calling `ComputeInstructionSetFlags` where we're adding this change, we add all the instruction sets provided through the command-line by means of `AddSupportedInstructionSet`. This method validates that the sets we're providing are supported by the chip we're compiling for. So, if say, we send `--instruction-set:avx512lol`, it will fail before reaching this code, so we're fine in that regard.